surfman/src/resolution.c

```diff
@@ -65,8 +65,27 @@ get_resolution_from_monitor(struct plugin *plugin, surfman_monitor_t monitor,
     internal_w = info->current_mode->htimings[SURFMAN_TIMING_ACTIVE];
     internal_h = info->current_mode->vtimings[SURFMAN_TIMING_ACTIVE];
 
-    if (internal_w && internal_w < 3000 && internal_h && internal_w < 3000)
-    {
+    /* If a component (internal_w or internal_h) is 0, fail.
+     * If a component is higher than 3000 and the other one is too small for 1920x1200, fail.
```

Comment:
I thought that would be fine with cropping (https://github.com/OpenXT/surfman/blob/master/plugins/drm/src/device-intel.c#L469). Although, this may be a problem within Surfman before the plugin. It is extremely unlikely than anything would display like that anyway and it would look horrible if it did with cropping too.